data-science-onramp/data-ingestion/setup.py

```diff
@@ -0,0 +1,163 @@
+import random
+import sys
+
+from time import time_ns
+
+from google.cloud import bigquery
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+
+from pyspark.sql.functions import UserDefinedFunction
+from pyspark.sql.types import IntegerType, StringType
+
+
+BUCKET_NAME = sys.argv[1]
+TABLE = "bigquery-public-data.new_york_citibike.citibike_trips"
+
+
+# START MAKING DATA DIRTY
+def random_select(items, weights):
+    '''Picks an item according to the cumulative weights'''
+    return random.choices(items, weights=weights, k=1)[0]
+
+
+def trip_duration(duration):
+    '''Converts trip duration to other units'''
+    seconds = str(duration) + " s"
+    minutes = str(float(duration) / 60) + " min"
+    hours = str(float(duration) / 3600) + " h"
+    return random_select([seconds, minutes, hours,
+                         str(random.randint(-1000, -1))],
+                         [0.3, 0.3, 0.3, 0.1])
+
+
+def station_name(name):
+    '''Replaces '&' with '/' with a 50% chance'''
+    return random.choice([name, name.replace("&", "/")])
+
+
+def user_type(user):
+    '''Manipulates the user type string'''
+    return random.choice([user, user.upper(), user.lower(),
+                          "sub" if user == "Subscriber" else user,
+                          "cust" if user == "Customer" else user])
+
+
+def gender(s):
+    '''Manipulates the gender string'''
+    return random.choice([s.upper(), s.lower(),
+                         s[0].upper() if len(s) > 0 else "",
+                         s[0].lower() if len(s) > 0 else ""])
+
+
+def convert_angle(angle):
+    '''Converts long and lat to DMS notation'''
+    degrees = int(angle)
+    minutes = int((angle - degrees) * 60)
+    seconds = int((angle - degrees - minutes/60) * 3600)
+    new_angle = str(degrees) + u"\u00B0" + \
```

Comment:
In Python3, all non-byte strings are unicode. You can safely change `u\u00b0` to `\u00b0`.